heltour/tournament/tests/test_login.py

```diff
@@ -20,13 +21,18 @@ def test_encode_decode_state(self, *args):
     @patch('heltour.tournament.oauth._encode_state', return_value='encodedstate')
     def test_oauth_redirect(self, *args):
         response = self.client.get(league_url('team', 'login'))
-        expected_oauth_url = 'https://oauth.lichess.org/oauth/authorize' + \
+        url = re.sub("&code_challenge=[0-9A-z-]{43}", "", response.url)
+        expected_oauth_url = 'https://lichess.org/oauth' + \
```

Comment:
```suggestion
        expected_oauth_url = ('https://lichess.org/oauth'
                             '?response_type=code'
                             '?response_type=code'
                             '&client_id=clientid'
                             '&client_id=heltour'
                             '&redirect_uri=http://testserver/auth/lichess/'
                             '&redirect_uri=https://testserver/auth/lichess/'
                             '&scope=email:read%20challenge:read%20challenge:write'
                             '&scope=email:read%20challenge:read%20challenge:write'
                             '&code_challenge_method=S256'
                             '&state=encodedstate'
                             '&state=encodedstate')
```
python will concat this strings for you.

I would also make an argument to use https://docs.python.org/3/library/urllib.parse.html#module-urllib.parse as you are looking at a URL. but not the end of the world